src/cli.erl

```diff
@@ -0,0 +1,69 @@
+-module(cli).
+-export([main/1]).
+-include("gradualizer.hrl").
+
+-spec main([string()]) -> ok | no_return().
+main([]) ->
+    print_usage();
+main(Args) -> handle_args(Args).
+
+-spec handle_args([string()]) -> no_return().
+handle_args(Args) ->
+    {Rest, Opts} = parse_opts(Args, []),
+    HasHelp = proplists:get_bool(help, Opts),
+    HasVersion = proplists:get_bool(version, Opts),
+    KeepGoing = not proplists:get_bool(stop_on_first_error, Opts),
+    {_, Exit} = if
+        HasHelp -> print_usage(), {true, 0};
+        HasVersion -> print_version(), {true, 0};
+        true -> lists:foldl(
+            fun(_, {false, Exit}) -> {false, Exit};
+            (File, {true, Exit}) ->
+                case gradualizer:type_check_file(File, Opts) of
+                    ok  -> {true, Exit};
+                    nok -> {KeepGoing, 1}
+                end
+            end, {true, 0}, Rest)
+    end,
+    halt(Exit).
+ 
+-spec get_ver(atom()) -> string().
+get_ver(App) ->
+    case lists:keyfind(App, 1, application:loaded_applications()) of
+        {_, _, Ver} ->
+            Ver;
+        false ->

```

Comment:
Why accept the false case here and hide the error in an empty string? Since we require the application to exist, it's better to crash IMO.

And why a generic get_ver when we only care about the gradualizer application? I would shorten all this to `{_, _, Ver} = lists:keyfind(gradualizer, 1, application:loaded_applications())`.

Reply:
Makes sense. A generic function was used because I found the example online. It's also easier for me to filter out the noise that way - inlining the function makes print_version/0 read less like prose. I've been getting the same feedback from my team and I'm not a convert yet.